pkg/apis/hive/v1alpha1/clusterdeployment_types.go

```diff
@@ -152,8 +162,49 @@ type ClusterDeploymentStatus struct {
 	// SyncPatches is the list of SyncStatus for patches that have been applied.
 	// +optional
 	SyncPatches []SyncStatus `json:"syncPatches,omitempty"`
+
+	// InstallerImage is the name of the installer image to use when installing the target cluster
+	// +optional
+	InstallerImage *string `json:"installerImage,omitempty"`
+
+	// Conditions includes more detailed status for the cluster deployment
+	// +optional
+	Conditions []ClusterDeploymentCondition `json:"conditions,omitempty"`
 }
 
+// ClusterDeploymentCondition contains details for the current condition of a cluster deployment
+type ClusterDeploymentCondition struct {
+	// Type is the type of the condition.
+	Type ClusterDeploymentConditionType `json:"type"`
+	// Status is the status of the condition.
+	Status corev1.ConditionStatus `json:"status"`
+	// LastProbeTime is the last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
+	// LastTransitionTime is the last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// Reason is a unique, one-word, CamelCase reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// Message is a human-readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// ClusterDeploymentConditionType is a valid value for ClusterDeploymentCondition.Type
+type ClusterDeploymentConditionType string
+
+const (
+	// ClusterImageSetNotFoundCondition is set when the ClusterImageSet referenced by the
+	// ClusterDeployment is not found.
+	ClusterImageSetNotFoundCondition ClusterDeploymentConditionType = "ClusterImageSetNotFound"
+
+	// InstallerImageResolutionFailedCondition is a condition that indicates whether the job
+	// to determine the installer image based on a release image was successful.
+	InstallerImageResolutionFailedCondition ClusterDeploymentConditionType = "InstallerImageResolutionFailed"
```

Comment:
Not in this PR, but between this and incoming preflight checks we should start up a document in the repo outlining the possible conditions and what they mean as a contract for us and UHC.